src/renderer/components/ColorPalette/ColorPalette.js

```diff
@@ -0,0 +1,164 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/**
+ * This is Reactjs functional component that create palette for change buttons color on keyboard
+ */
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { withStyles } from "@material-ui/core/styles";
+import Paper from "@material-ui/core/Paper";
+import ColorButtonsArea from "./ColorButtonsArea";
+import PickerColorButton from "./PickerColorButton";
+
+ColorPalette.propTypes = {
+  classes: PropTypes.object.isRequired,
+  onColorSelect: PropTypes.func.isRequired,
+  palette: PropTypes.array.isRequired,
+  onColorPick: PropTypes.func.isRequired,
+  disabled: PropTypes.bool.isRequired,
+  selected: PropTypes.number.isRequired
+};
+
+const styles = theme => ({
+  root: {
+    display: "flex",
+    position: "fixed",
+    left: 0,
+    right: 0,
+    bottom: 0,
+    height: 150,
+    justifyContent: "center",
+    alignItems: "flex-start"
+  },
+  palette: {
+    display: "flex",
+    alignItems: "center",
+    padding: 10,
+    [theme.breakpoints.down("sm")]: {
+      width: 470
+    }
+  }
+});
+
+/**
+ * Use to reduce the amount of code
+ * @param {object} color Object with keys that defining colors using the Red-green-blue-alpha (RGBA) model
+ */
+const setColorTamplate = color => ({
+  r: color.r,
+  g: color.g,
+  b: color.b,
+  rgb: `rgb(${color.r}, ${color.g}, ${color.b})`
+});
+
+/**
+ * Reactjs functional component that create palette for change buttons color on keyboard
+ * @param {object} classes Property that sets up CSS classes that adding to HTML elements
+ * @param {function} onColorSelect Callback function from Editor component for change color of buttons in keyboard. Parameter is index of color button in palette (from 0 to 15)
+ * @param {array} palette Array of colors. Format [{r: 200, g: 200, b: 200, rgb: "rgb(200, 200, 200)"}, ...]
+ * @param {function} onColorPick Callback function from Editor component for change color of buttons in ColorPalette. Parameters are: first - index of color button in palette (from 0 to 15), second - index of color (r: from 0 to 255), third - index of color (g: from 0 to 255), fourth - index of color (b: from 0 to 255)
+ * @param {boolean} disabled Property that disable component
+ * @param {number} selected Number of selected color button in palette (from 0 to 15)
+ */
+function ColorPalette(props) {
+  const {
+    classes,
+    onColorSelect,
+    palette,
+    onColorPick,
+    disabled,
+    selected
+  } = props;
+
+  /**
+   * This is Hook that lets add React state "indexFocusButton" to functional components
+   * @param {object} [initialState=selected] - Sets initial state for "indexFocusButton"
+   */
+  const [indexFocusButton, setIndexFocusButton] = useState(selected);
+
+  /**
+   * Use variable to reduce the amount of code
+   */
+  const colorForFocusButton = {
```

Comment:
Why are we creating this variable at all vs just using `...palette[selected]` where you are using `...colorForFocusButton`?